scripting/src/main/java/org/infinispan/scripting/impl/LocalRunner.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.scripting.impl;
+
+import java.util.concurrent.Future;
+
+import javax.script.CompiledScript;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.infinispan.commons.util.concurrent.NoOpFuture;
+
+public class LocalRunner implements ScriptRunner {
+   public static final LocalRunner INSTANCE = new LocalRunner();
+
+   private LocalRunner() {
+   }
+
+   @Override
+   public <T> Future<T> runScript(ScriptingManagerImpl scriptManager, String scriptName, ScriptMetadata metadata, CacheScriptBindings bindings) throws ScriptException {
+      CompiledScript compiled = scriptManager.compiledScripts.get(scriptName);
+      if (compiled != null) {
+         T result = (T) compiled.eval(bindings);
+         return new NoOpFuture<T>(result);
+      } else {
+         ScriptEngine engine = scriptManager.getEngineForScript(scriptName, metadata);
+         T result = (T) engine.eval(scriptManager.scriptCache.get(scriptName), bindings);
```

Comment:
EXEC only allows you to execute scripts, and you can only pass parameters/values. You will need to have a specific role to be able to access the script cache and read/write/update the scripts. Obviously if authentication/authorization is not enabled globally this would be pointless, so I want to make these internal caches not available over the remoting protocols by default.